src/main/java/hudson/plugins/sloccount/SloccountBuildAction.java

```diff
@@ -101,11 +115,33 @@ SloccountBuildAction getPreviousAction(){
         return null;
     }
 
-    public AbstractBuild<?,?> getBuild(){
+    public Run<?,?> getBuild(){
         return this.build;
     }
 
     public Object getTarget() {
         return this.result;
     }
+    
+    @Override
+    public void onLoad(Run<?,?> r)
+    {
+        this.build = r;
+        if( this.projectActions == null )
+        {
+            this.projectActions = new ArrayList<>();  
```

Comment:
This collection type is not thread-safe. Without synchronization such logic will likely fail at some point

Reply:
Still does not clarify the thread safety concern. What does block you from making methods more reliable if you really want to use internal caching?